_blog/2024-02-09-using-clerk-in-rust.mdx

```diff
@@ -0,0 +1,1014 @@
+    .await;
+
+    if update_query.is_err() {
+        return HttpResponse::InternalServerError().json(serde_json::json!({
+            "status":"FAILED",
+            "message":"Failed to update the issue"
+        }));
+    }
+
+    HttpResponse::Ok().json(serde_json::json!({
+        "status": "SUCCESS",
+        "message":"Updated successfully"
+    }))
+}
+
+```
+
+### Delete issue by `id` endpoint
+
+This endpoint is for deleting the issue by `id` from the database:
+
+```rust
+#[delete("/issue/{issue_id}")]
+async fn delete_issue(
+    state: web::Data<AppState>,
+    path: web::Path<i32>,
+    req: HttpRequest,
+) -> impl Responder {
+    let srv_req = ServiceRequest::from_request(req);
+
+    let claim = match clerk_authorize(&srv_req, &state.client, true).await {
+        Ok(value) => value.1,
+        Err(_) => {
+            return HttpResponse::Unauthorized().json(serde_json::json!({
+                "status":"Failed",
+                "message":"Unauthorized"
+            }));
+        }
+    };
+
+    let issue_id = path.into_inner();
+
+    let query: Result<Issue, sqlx::Error> = sqlx::query_as("SELECT * FROM issues WHERE id=$1")
+        .bind(issue_id)
+        .fetch_one(&state.pool)
+        .await;
+
+    match query {
+        Ok(issue) => {
+            if issue.author != claim.sub {
+                return HttpResponse::Unauthorized().json(serde_json::json!({
+                    "status":"FAILED",
+                    "message":"No authorized to delete the issue."
+                }));
+            }
+        }
+        Err(_) => {
+            return HttpResponse::NotFound().json(serde_json::json!({
+                "status":"FAILED",
+                "message":"Issue does not exist."
+            }));
+        }
+    }
+
+    let delete_query: Result<Issue, sqlx::Error> = sqlx::query_as("DELETE FROM issues WHERE id=$1")
+        .bind(issue_id)
+        .fetch_one(&state.pool)
+        .await;
+
+    if delete_query.is_err() {
+        return HttpResponse::InternalServerError().json(serde_json::json!({
+            "status":"FAILED",
+            "message":"Failed to delete the issue"
+        }));
+    }
+
+    HttpResponse::Ok().json(serde_json::json!({
+        "status": "SUCCESS",
+        "message":"Deleted successfully"
+    }))
+}
+
+```
+
+### Get user by `id`
+
+We need on last endpoint which can fetch the user details by their `user_id`, this API will be used to fetch issue author metadata from Clerk:
+
+```rust
+#[get("/user/{user_id}")]
+async fn get_user_by_id(state: web::Data<AppState>, path: web::Path<String>) -> impl Responder {
+    let user_id = path.into_inner();
+    let Ok(user) = User::get_user(&state.client, &user_id).await else {
+        return HttpResponse::InternalServerError().json(serde_json::json!({
+            "status": "FAILED",
+            "message": "Unable to retrieve all users",
+        }));
+    };
+
+    HttpResponse::Ok().json(Into::<UserModel>::into(user))
```

Comment:
Mention in the article text that this was adapted from existing code in the template, so that the reader konws where `UserModel` is from.